R/config.R

```diff
@@ -185,48 +172,203 @@ print.snowflake_redacted <- function(x, ...) {
+#'
+#' @keywords internal
+load_config <- function(name = NULL, config_dir = default_config_dir()) {
+  # Initialize result structure
+  result <- list(
+    connections = list(),
+    connection_name = NULL,
+    connection_file = NULL
+  )
+
+  # File paths
+  config_toml <- file.path(config_dir, "config.toml")
+  connections_toml <- file.path(config_dir, "connections.toml")
+
+  # Step 1: Resolve connection name with proper precedence:
+  # 1. Explicit name parameter
+  # 2. SNOWFLAKE_DEFAULT_CONNECTION_NAME environment variable
+  # 3. default_connection_name from config.toml
+  # 4. "default" section in connections.toml
+
+  # Set from explicit parameter if provided
+  if (!is.null(name)) {
+    result$connection_name <- name
+  } else {
+    # Check environment variable
+    env_connection_name <- Sys.getenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "")
+    if (nzchar(env_connection_name)) {
+      result$connection_name <- env_connection_name
+    }
+  }
+
+  # Step 2: Load configuration files
+  has_config_toml <- file.exists(config_toml)
+  has_connections_toml <- file.exists(connections_toml)
+
+  # Load config.toml
+  if (has_config_toml) {
+    config <- RcppTOML::parseTOML(config_toml, fromFile = TRUE)
+
+    # If no connection name yet, get from config.toml
+    if (
+      is.null(result$connection_name) &&
+        !is.null(config$default_connection_name)
+    ) {
+      result$connection_name <- config$default_connection_name
+    }
+
+    # Extract connections section if it exists
+    if (!is.null(config$connections)) {
+      result$connections <- config$connections
+      result$connection_file <- config_toml
+    }
+  }
+
+  # Load connections.toml (takes precedence if both exist)
+  if (has_connections_toml) {
+    connections <- RcppTOML::parseTOML(connections_toml, fromFile = TRUE)
+
+    # If both files exist, inform user we're using connections.toml
+    if (has_config_toml && length(connections) > 0) {
+      cli::cli_inform(c(
+        "!" = "Both {.file connections.toml} and {.file config.toml} exist. Using {.file connections.toml}."
+      ))
+
+      # Validate that connection name from config.toml exists in connections.toml
+      if (
+        !is.null(result$connection_name) &&
+          result$connection_name != "default" &&
+          is.null(connections[[result$connection_name]])
+      ) {
+        cli::cli_abort(c(
+          "{.field default_connection_name} is set to {.str {result$connection_name}} in {.file config.toml},
+          but the connection does not exist in {.file connections.toml}.",
+          i = "Try defining a {.field [{result$connection_name}]} section in {.file connections.toml}."
+        ))
+      }
+    }
+
+    # Set connections in result
+    result$connections <- connections
+    result$connection_file <- connections_toml
+
+    # Set default connection name if not already set and "default" exists
+    if (is.null(result$connection_name) && !is.null(connections[["default"]])) {
```

Comment:
Setting the default to "default" should always happen, not only when processing `connections.toml`.